docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,247 @@
+!!!
+
+Please follow [this guide](https://facebook.github.io/react-native/docs/getting-started) for how to create a new React Native application if you need one. Also ensure you have followed the [Getting Started](Getting Started/Installation) and have `SmartDeviceLink` installed on the native side. 
+
+To install SDL into your React Native app, you will need to follow [this guide](https://facebook.github.io/react-native/docs/native-modules-ios) to integrate the SDL library into your application using React Native's Native Modules feature.
+
+## Getting Started
+This guide is not meant to walk you through how to make a React Native app but help you integrate SDL into an existing application. We will show you a basic example of how to communicate between your app's JavaScript code and SDL's native Obj-C code. For more advanced features, please refer to the React Native documentation linked above.
+
+## Integration Basics
+Native API methods are not exposed automatically to JavaScript. This means you must expose methods you wish to use from SDL to your React Native app. You must implement the `RCTBridgeModule` protocol into a bridge class (see below for an example). Please follow [SmartDeviceLink Integration Basics](Getting Started/Integration Basics) for the basic setup of a native SDL `ProxyManager` class that your bridge code will communicate with. This is the necessary starting point in order to continue with this example. Please make sure you also set up a simple UI with buttons and some text on the SDL side.
+
+### Creating the RCTBridge
+To create a native module you must implement the `RCTBridgeModule` protocol. Update your `ProxyManager` to include `RCTBridgeModule`.
+
+##### Objective-C
+###### ProxyManager.h
+```objc
+#import <React/RCTBridgeModule.h>
+
+@interface ProxyManager : NSObject <RCTBridgeModule>
+
+<#...#>
+
+@end
+```
+
+###### ProxyManager.m
+An `RCT_EXPORT_MODULE()` macro must be added to the implementation file to expose the class to React Native.
+```objc
+@implementation ProxyManager
+
+RCT_EXPORT_MODULE();
+<#...#>
+
+@end
+```
+
+##### Swift
+First, you must add  `#import "React/RCTBridgeModule.h"` to your `Bridging Header` before you move forward. When creating a Swift application and importing Objective-C code, Xcode should ask if it should create this header file for you. You must include this bridging header for your React Native app to work. You can create this file manually as well. 
+
+```swift
+@objc(ProxyManager)
+
+class ProxyManager: NSObject {
+
+<#...#>
+
+}
+```
+
+Next, to expose the Swift class to React Native you must create an Objective-C file in order to use React Native macros.
+
+##### Objective-C
```

Comment:
This is confusing. Is it part of the Swift code that has to be added?

Reply:
yes it is. it is the Bridge class needed to talk to javascript